components/uniset/src/conversions.rs

```diff
@@ -0,0 +1,123 @@
+use crate::utils::deconstruct_range;
+use crate::UnicodeSet;
+use std::{
+    convert::TryFrom,
+    ops::{Range, RangeBounds, RangeFrom, RangeFull, RangeInclusive, RangeTo, RangeToInclusive},
+};
+
+fn try_from_range_impl(range: impl RangeBounds<char>) -> Result<UnicodeSet, (u32, u32)> {
+    let (from, till) = deconstruct_range(range);
+    if from < till {
+        let set = vec![from, till];
+        Ok(UnicodeSet::try_from(set).unwrap())
+    } else {
+        Err((from, till))
+    }
+}
+
+impl TryFrom<Range<char>> for UnicodeSet {
```

Comment:
Does `impl TryFrom<&dyn RangeBounds<char>>` work?

Reply:
Unfortunately as `RangeBounds` has a provided method `contains` which contains a generic, so rust compiler throws an error about safety. There are ways to get around this but it makes the code more confusing as you would have to create a object-safe wrapper trait around `RangeBounds`, and then you need to use that instead of `RangeBounds` throughout the code and etc. I asked about this in one of the rust discussion chat rooms, and it seems that it would just be easiest to include implementations for all of the `Range` structs.